src/core/utils/parse_cmdline.cpp

```diff
@@ -302,12 +302,17 @@ enum HexStringParseMode
 
 static Error ParseHexString(const char *&aString, uint16_t &aSize, uint8_t *aBuffer, HexStringParseMode aMode)
 {
-    Error  error        = kErrorNone;
-    size_t stringLength = strlen(aString);
-    size_t expectedSize = (stringLength + 1) / 2;
-    size_t parsedSize   = 0;
+    Error  error      = kErrorNone;
+    size_t parsedSize = 0;
+    size_t stringLength;
+    size_t expectedSize;
     bool   skipFirstDigit;
 
+    VerifyOrExit(aString != nullptr, error = kErrorInvalidArgs);
```

Comment:
When can this be nullptr?

Reply:
With #6767, the model changed to now rely on `ParseAs()` to return error when no argument exists. One example:
https://github.com/openthread/openthread/blob/8967d62ead1679c5cdd0e94e194826fee8f8f7b8/src/cli/cli.cpp#L2861